Tasks/CloudFormationCreateOrUpdateStack/TaskParameters.ts

```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { AWSConnectionParameters, buildConnectionParameters } from 'Common/awsConnectionParameters'
+import { defaultTimeoutInMins } from 'Common/cloudformationutils'
+import fs = require('fs')
+import tl = require('vsts-task-lib/task')
+
+export const fileSource: string = 'file'
+export const urlSource: string = 'url'
+export const s3Source: string = 's3'
+export const usePreviousTemplate: string = 'usePrevious'
+
+export const maxRollbackTriggers: number = 5
+export const maxTriggerMonitoringTime: number = 180
+
+export const loadTemplateParametersFromFile: string = 'file'
+export const loadTemplateParametersInline: string = 'inline'
+
+export const ignoreStackOutputs: string = 'ignore'
+export const stackOutputsAsVariables: string = 'asVariables'
+export const stackOutputsAsJson: string = 'asJSON'
+
+export interface TaskParameters {
+    awsConnectionParameters: AWSConnectionParameters
+    stackName: string
+    templateSource: string
+    templateFile: string
+    s3BucketName: string
+    s3ObjectKey: string
+    templateUrl: string
+    templateParametersSource: string
+    templateParametersFile: string
+    templateParameters: string
+    useChangeSet: boolean
+    changeSetName: string
+    description: string
+    autoExecuteChangeSet: boolean
+    capabilityIAM: boolean
+    capabilityNamedIAM: boolean
+    capabilityAutoExpand: boolean
+    roleARN: string
+    notificationARNs: string[]
+    resourceTypes: string[]
+    tags: string[]
+    monitorRollbackTriggers: boolean
+    monitoringTimeInMinutes: number
+    rollbackTriggerARNs: string[]
+    onFailure: string
+    warnWhenNoWorkNeeded: boolean
+    outputVariable: string
+    captureStackOutputs: string
+    captureAsSecuredVars: boolean
+    timeoutInMins: number
+}
+
+export function buildTaskParameters(): TaskParameters {
+    const parameters: TaskParameters = {
+        awsConnectionParameters: buildConnectionParameters(),
+        stackName: tl.getInput('stackName', true),
+        templateSource: tl.getInput('templateSource', true),
+        templateFile: undefined,
+        s3BucketName: undefined,
+        s3ObjectKey: undefined,
+        templateUrl: undefined,
+        templateParametersSource: tl.getInput('templateParametersSource', true),
+        templateParametersFile: undefined,
+        templateParameters: undefined,
+        useChangeSet: tl.getBoolInput('useChangeSet', false),
+        changeSetName: undefined,
```

Comment:
Is it possible to pull line 137 up here?

Reply:
Nope because it uses `useChangeSet`, if only this were Kotlin